_projects/decrease_docker_image_size.md

```diff
@@ -0,0 +1,76 @@
+---
+collaborating_projects:
+  - docker-coala-base
+desc: "Find creative ways to decrease the coala Docker image size"
+issues:
+  - https://github.com/coala/docker-coala-base/issues/55
+difficulty: "hard"
+initiatives:
+  - GCI
+markdown: decrease_docker_image_size.md
+mentors:
+  - yukiisbored
+name: "Decrease Docker image size"
+tags:
+  - coala
+  - docker
+  - linux
+---
+
+## Task
+
+Submit a Pull Request to the [docker-coala-base][repo] project which decreases
+the image size.
+
+## Objective
+
+This task is considerably hard which requires creativity and knowledge for
+Docker and Linux to decrease the Docker Image size.
+
+## Background
+
+The Docker image for coala is huge (around ~2G!) which makes it a huge drawback
+to be used especially for those who are with limited Internet access.
+
+Decreasing it isn't easy because a lot of linters requires a lot of dependencies
+and some weird edge cases as well.
+
+Since this is Docker where images are compose of layers, deleting stuff from
+other layers (like the openSUSE base image) doesn't change much.
+
+## Finding bloat
+
+To find bloat, we can start by seeing the size generated by each layer. We can
+do this by running `docker history <image name>` which will give us each layers'
+size.
+
+Some of these layers are probably outdated since `docker history` also shows
+previous versions as well, so make sure you're looking at a layer which is a
+part of the latest image.
+
+Finding the actual files within the layer can be challenging and requires
+knowing what the command which makes the layer does.
+
+For example, when installing a program it may contain some documentations since
+this is a Docker image which sole purpose is to run coala the documentation
+isn't really helpful or even important.
+
+## Removing bloat
+
+Removing bloat for a layer can be done just by adding another command to the
+layer's command. This should be done **not** by adding another `RUN` command but
+appending it's `RUN` command with `&&`.
+
+Example:
+```Dockerfile
+RUN zypper in kirisame && \
+    rm -rf /usr/share/man/man1/kirisame.gz \
+           /usr/share/man/man6/kirisamectl.gz
+```
+
+## Submit a PR
+
+If you've decreased any amount of the Docker Image, please create a PR and
+submit it to the [repo][repo].
```

Comment:
Anyone have any ideas how to improve this :P

Reply:
you can add link to http://api.coala.io/en/latest/Developers/Writing_Good_Commits.html

so that they send valid commits

Reply:
I think that's a bit redundant since it's already on the PR template...